CMakeLists.txt

```diff
@@ -328,35 +331,54 @@ if(CMAKE_SYSTEM_NAME MATCHES "Darwin|DragonFly|FreeBSD|Linux|NetBSD|OpenBSD")
   list(APPEND uv_test_libraries util)
 endif()
 
-add_library(uv SHARED ${uv_sources})
-target_compile_definitions(uv PRIVATE ${uv_defines} BUILDING_UV_SHARED=1)
-target_compile_options(uv PRIVATE ${uv_cflags})
-target_include_directories(uv PRIVATE include src)
-target_link_libraries(uv ${uv_libraries})
+if(CYGWIN)
+  list(APPEND uv_defines _GNU_SOURCE)
+  list(APPEND uv_sources
+       src/unix/cygwin.c
+       src/unix/bsd-ifaddrs.c
+       src/unix/no-fsevents.c
+       src/unix/no-proctitle.c
+       src/unix/posix-hrtime.c
+       src/unix/posix-poll.c
+       src/unix/procfs-exepath.c
+       src/unix/sysinfo-loadavg.c
+       src/unix/sysinfo-memory.c)
+endif()
+
+add_library(uv_shared SHARED ${uv_sources})
+target_compile_definitions(uv_shared PRIVATE ${uv_defines} BUILDING_UV_SHARED=1)
+target_compile_options(uv_shared PRIVATE ${uv_cflags})
+target_include_directories(uv_shared PRIVATE include src)
+target_link_libraries(uv_shared ${uv_libraries})
+set_target_properties(uv_shared PROPERTIES OUTPUT_NAME "uv")
+if(MSVC AND LIBUV_ENABLE_STATIC)
+  set_target_properties(uv_shared PROPERTIES IMPORT_SUFFIX "_import.lib")
+endif()
```

Comment:
This condition makes the name of the `.lib` import library different depending on how libuv was configured.  That makes it hard for dependents to know how to link to the library.

Some projects use the convention `uv.{dll,lib}` and `libuv.lib` for the shared and static variants, respectively.